src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,454 @@
+                    .append(entrypointCounter).append(":").append(e.getFieldName());
+                    currentPath.append(entrypointAlias.toString());
+                }
+                else{
+                    // Get full list entrypoint (No argument values)
+                    e = endpoint.getEntrypoint(sgpType,GraphQLEntrypointType.FULL);
+                    final StringBuilder entrypointAlias = new StringBuilder("ep_full")
+                    .append(entrypointCounter).append(":").append(e.getFieldName());
+                    currentPath.append(entrypointAlias.toString());
+                }
+ 
+                // If entrypoint has arguments then add them to the currentPath, argumentValues and argumentDefinitions
+                final Map<String,String> entrypointArgDefs = e.getArgumentDefinitions();
+                if(entrypointArgDefs != null && !entrypointArgDefs.isEmpty()){
+                    currentPath.append("(");
+                    for(final String argName : new TreeSet<String>(entrypointArgDefs.keySet())){
+                        final String variableName = "var"+variableCounter;
+                        currentPath.append(argName).append(":$").append(variableName).append(",");
+                        argumentDefinitions.put(variableName,entrypointArgDefs.get(argName));
+                        if(sgpArguments.containsKey(argName)) {
+                            argumentValues.put(variableName,literalToJsonValue(sgpArguments.get(argName)));
+                        }
+                        else{
+                            argumentValues.put(variableName,JsonNull.instance);
+                        }
+                        ++variableCounter;
+                    }
+                    final int lastComma = currentPath.lastIndexOf(",");
+                    currentPath.deleteCharAt(lastComma);
+                    currentPath.append(")");
+                }
+
+                // Start adding fields for nested object(s) using recursion
+                currentPath.append("/");
+                addSgp(fieldPaths,subgraphPatterns,connectors,config,endpoint,
+                    n,currentPath.toString(),sgpType);
+                ++entrypointCounter;
+            }
+        }
+
+        return new GraphQLQueryImpl(fieldPaths, argumentValues, argumentDefinitions);
+    }
+
+
+    /**
+     * Recursive function used to add fields from the TPs in a given SGP
+     */
+    protected void addSgp(final TreeSet<String> fieldPaths, final Map<Node,Set<TP>> subgraphPatterns, 
+            final Map<Integer,Node> connectors, final GraphQL2RDFConfiguration config, final GraphQLEndpoint endpoint, 
+            final Node subgraphNode, final String currentPath, final String sgpType){
+
+        // Necessary id field present in all objects used to indentify the GraphQL object
+        fieldPaths.add(new StringBuilder(currentPath).append("id_").append(sgpType).append(":id").toString());
```

Comment:
There's no need to use `StringBuilder`s everywhere, they are mainly useful when you're making a string on a loop, not necessarily when creating just one string in one line.